v0.41.0 regression: incorrect damage attribution and missing player rounds in overtime demos

## Description

After upgrading from demoparser2 **0.40.3** to **0.41.0**, we observed incorrect ADR (Average Damage per Round) calculations on an overtime match. The issue manifests as:

1. **Extra damage events with null victim** — events where `attacker_name` is set but `victim_name` / `victim_steamid` are null, inflating total damage
2. **Incorrect side attribution** — enemy damage events are misclassified (attacker and victim on different sides are reported as same-side), causing damage to be wrongly categorized as team damage
3. **Missing player rounds** — some players only appear in a subset of rounds (e.g. 16/30 instead of 30/30), leading to incorrect per-round stats

All three issues are absent in v0.40.3 on the same demo file.

## Reproduction

**Demo file:** `003804544737319322155_1902595074.dem.bz2`
**Download:** `http://replay390.valve.net/730/003804544737319322155_1902595074.dem.bz2`
**Match:** de_nuke, MR12 + MR3 overtime, 30 rounds, final score 16-14

### Minimal reproduction (using [awpy](https://github.com/pnxenopoulos/awpy))

```python
from awpy import Demo
from awpy.stats import adr

demo = Demo("003804544737319322155_1902595074.dem")
demo.parse()

# Compare ADR with and without team damage filtering
print(adr(demo))                    # default (includes team damage)
print(adr(demo, team_dmg=True))     # excludes team damage
```

### Expected vs actual

#### Example player A (played all 30 rounds)

| Metric | v0.40.3 (correct) | v0.41.0 (broken) |
|---|---|---|
| Total damage | 3042 | 3135 (+93 ghost damage) |
| ADR (default) | 101.4 | 104.5 |
| Team/self damage | 9 | 441 (enemy dmg misattributed) |
| ADR (team dmg excluded) | 101.1 | 89.8 |

The real ADR is ~101 (confirmed by in-game scoreboard and Leetify).

#### Example player B (played all 30 rounds)

| Metric | v0.40.3 (correct) | v0.41.0 (broken) |
|---|---|---|
| Rounds present | 30 | **16** (missing rounds 1-12 and 29-30) |
| ADR (default) | 110.8 | 105.2 (wrong denominator) |

#### Full comparison (all 10 players, anonymized)

```
Player  Rounds | v0.40.3 ADR   v0.41.0 ADR  Diff  | v0.40.3 ADR*  v0.41.0 ADR*  Diff
---------------------------------------------------------------------------------------
A       30     |      101.40      104.50   +3.10   |      101.10       89.80   -11.30
B       30     |       73.07       73.17   +0.10   |       72.03       67.40    -4.63
C       30     |      101.87      105.57   +3.70   |      101.87       93.27    -8.60
D       30     |       62.13       66.80   +4.67   |       61.83       51.37   -10.46
E       30     |       85.57       85.57   +0.00   |       84.87       84.87    +0.00
F       30     |       28.47       28.47   +0.00   |       28.47       28.47    +0.00
G       30     |       96.70       96.70   +0.00   |       95.50       95.50    +0.00
H       30     |       51.80       51.80   +0.00   |       51.47       51.47    +0.00
I       30     |       72.17       73.23   +1.06   |       72.17       65.83    -6.34
J       30     |      110.80      105.19   -5.61   |      110.80      105.19    -5.61

* ADR with team damage excluded (filtering attacker_side != victim_side)
```

Players E–H are unaffected — the bug does not impact all players equally.

## Environment

- **demoparser2**: 0.41.0 (regression) vs 0.40.3 (works correctly)
- **Python**: 3.14
- **OS**: macOS (ARM64) and Linux (aarch64)
- **awpy**: 2.0.2 (used for ADR calculation, but the underlying data comes from demoparser2)

--- a/src/parser/src/second_pass/collect_data.rs
+++ b/src/parser/src/second_pass/collect_data.rs
@@ -1079,6 +1079,60 @@ impl<'a> SecondPassParser<'a> {
                 }
             }
         }
+        if entity.entity_type == EntityType::PlayerController {
+            let team_num = match self.prop_controller.special_ids.teamnum {
+                Some(team_num_id) => match self.get_prop_from_ent(&team_num_id, entity_id) {
+                    Ok(Variant::U32(team_num)) => Some(team_num),
+                    Ok(_) => return Err(DemoParserError::IncorrectMetaDataProp),
+                    Err(_) => None,
+                },
+                _ => None,
+            };
+            let name = match self.prop_controller.special_ids.player_name {
+                Some(id) => match self.get_prop_from_ent(&id, entity_id) {
+                    Ok(Variant::String(name)) => Some(name),
+                    Ok(_) => return Err(DemoParserError::IncorrectMetaDataProp),
+                    Err(_) => None,
+                },
+                _ => None,
+            };
+            let steamid = match self.prop_controller.special_ids.steamid {
+                Some(id) => match self.get_prop_from_ent(&id, entity_id) {
+                    Ok(Variant::U64(sid)) => Some(sid),
+                    Ok(_) => return Err(DemoParserError::IncorrectMetaDataProp),
+                    Err(_) => None,
+                },
+                _ => None,
+            };
+            let player_entid = match self.prop_controller.special_ids.player_pawn {
+                Some(id) => match self.get_prop_from_ent(&id, entity_id) {
+                    Ok(Variant::U32(handle)) => Some((handle & 0x7FF) as i32),
+                    Ok(_) => return Err(DemoParserError::IncorrectMetaDataProp),
+                    Err(_) => None,
+                },
+                _ => None,
+            };
+            if let Some(e) = player_entid {
+                if e != PLAYER_ENTITY_HANDLE_MISSING && steamid != Some(0) && team_num != Some(SPECTATOR_TEAM_NUM) {
+                    match self.should_remove(steamid) {
+                        Some(eid) => {
+                            self.players.remove(&eid);
+                        }
+                        None => {}
+                    }
+                    self.players.insert(
+                        e,
+                        PlayerMetaData {
+                            name,
+                            team_num,
+                            player_entity_id: player_entid,
+                            steamid,
+                            controller_entid: Some(*entity_id),
+                        },
+                    );
+                }
+            }
+        }
         Ok(())
     }
     pub fn should_remove(&self, steamid: Option<u64>) -> Option<i32> {

